src/sentry/api/event_search.py

```diff
@@ -559,7 +561,8 @@ def convert_search_filter_to_snuba_query(search_filter):
             env_conditions.append(["environment", operator, None])
 
         if _envs:
-            env_conditions.append(["environment", "IN", list(_envs)])
```

Comment:
@markstory I have no idea why this doesn't work 😢

Reply:
You should also check that issue search still works, as this code is re-used there.

Reply:
From my slack conversations with @markstory .

```python
# this works
env_conditions.append(["environment", "IN", set(_envs)])

# this does not work
env_conditions.append(["environment", "IN", list(_envs)])
```

The generated snuba queries are as follows:

```python
# env_conditions.append(["environment", "IN", set(_envs)])
{
    'orderby': [u'-count'],
    'consistent': False,
    'aggregations': [('count()', None, 'count')],
    'dataset': 'events',
    'project': [2, 3],
    'from_date': '2019-07-24T19:24:30.886676',
    'limit': 9,
    'to_date': '2019-10-22T19:24:30.886676',
    'conditions': [[['environment', 'IN', set([u'staging'])]],
                   ['environment', 'IS NOT NULL', None], ('project_id',
                   'IN', [3, 2])],
    'groupby': ['environment'],
    }

# env_conditions.append(["environment", "IN", list(_envs)])
{
    'orderby': [u'-count'],
    'consistent': False,
    'aggregations': [('count()', None, 'count')],
    'dataset': 'events',
    'project': [2, 3],
    'from_date': '2019-07-24T19:22:51.465087',
    'limit': 9,
    'to_date': '2019-10-22T19:22:51.465087',
    'conditions': [[['environment', 'IN', [u'tags[staging]']]],
                   ['environment', 'IS NOT NULL', None], ('project_id',
                   'IN', [3, 2])],
    'groupby': ['environment'],
    }

```